Crashsite mission

wersal454 opened the pull request:
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [x] Enhancement

### What have you changed and why?
Information:

WIP
added a new mission to retrive valuable data from crashed object
Ideally object is whatever you define in template (wip)    

### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [ ] No
2. [x] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************

wersal454 committed:
fn_effects

wersal454 committed:
sonic(not the hedgehog) sound

wersal454 committed:
Update CfgFunctions.hpp

wersal454 committed:
Update CfgSounds.hpp

wersal454 committed:
crashsite mission

wersal454 committed:
fix fn_effects

Silence-112 commented on A3A/addons/core/functions/Missions/fn_effects:
```diff
@@ -0,0 +1,84 @@
+	
+	setAperture 10;
+
+	playSound3D ["x\A3A\addons\core\Sounds\Misc\Sonic.ogg", _object, false, getPosASL _object, 5, 1, 5000];
```
You should be able to use QPATHTOFOLDER since you #include script_component.hpp at the top
```suggestion
	playSound3D [QPATHTOFOLDER(Sounds\Misc\Sonic.ogg), _object, false, getPosASL _object, 5, 1, 5000];
```

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_effects

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

wersal454 committed:
mission request.sqf

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 committed:
Update fn_effects

wersal454 committed:
temporary update

Squad logic is somewhat broken + I can't figure out checks for vehicle types so I temporary commented them out.

wersal454 committed:
temporary update

wersal454 committed:
temporary update for testing

wersal454 commented:
Made an update with latest additions & changes. Can't figure out squad logic(there are tons of errors) + can't figure out vehicle type checks, might be forced to split this mission into 2 functions(1 for regular vehicles & 1 for satellite like vehicles)

wersal454 committed:
Merge branch 'unstable' into crashsite-mission

wersal454 commented:
also effects function, needs to be moved in dedicated directory, scrt has one

wersal454 commented:
You can check the mission via command ["LOG"] remoteExec ["A3A_fnc_missionRequest", 2];

Silence-112 commented on A3A/addons/core/CfgFunctions.hpp:
```diff
@@ -403,6 +403,7 @@ class CfgFunctions
             class LOG_Airdrop {};
             class LOG_Helicrash {};
             class missionRequest {};
+            class fn_effects {};
```
file name is fn_effects, the config however does not need the fn_ part.
Also, rename fn_effects to fn_LOG_Crashsite_Effects
```suggestion
            class LOG_Crashsite_Effects {};
```

wersal454 replied:
shoud be called something else, or simply moved to
![image](https://github.com/SilenceIsFatto/A3-Antistasi-Ultimate/assets/29953387/75ac2ff2-234b-439b-888f-bbe9dbbcae71)

Silence-112 replied:
that'll work too

wersal454 committed:
oups

wersal454 committed:
moving fn_effects

wersal454 committed:
Update CfgFunctions.hpp

wersal454 committed:
Update CfgFunctions.hpp

wersal454 committed:
Delete A3A/addons/core/functions/Missions/fn_effects

wersal454 committed:
Update Stringtable.xml

wersal454 committed:
Update Stringtable.xml

wersal454 committed:
Update fn_LOG_Helicrash.sqf

We want to avoid weird situations, where heli or box may collide with terrain objects. So I added additional terrain objects to destruction and deletion.

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 committed:
Split crashsite mission into 2

wersal454 committed:
Update CfgFunctions.hpp

wersal454 committed:
Update fn_effects

enabled the check for droppod vehicle

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 committed:
Update fn_LOG_Crashsite_Satelite.sqf

wersal454 committed:
typo

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 committed:
Update fn_LOG_Crashsite_Satelite.sqf

wersal454 committed:
Update Vanilla_AI_NATO_Arid.sqf

wersal454 committed:
Update fn_compileMissionAssets.sqf

wersal454 committed:
Update fn_initVarServer.sqf

wersal454 committed:
Update Vanilla_AI_CSAT_Arid.sqf

wersal454 committed:
Update Vanilla_AI_CSAT_Arid.sqf

wersal454 committed:
Update Vanilla.hpp

wersal454 committed:
Update Vanilla.hpp

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 committed:
Update fn_LOG_Crashsite_Satelite.sqf

wersal454 committed:
Update fn_effects

wersal454 committed:
typo

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 committed:
Update fn_LOG_Crashsite_Satelite.sqf

wersal454 commented:
Majority of the work is done, now we need a way to select the mission
```sqf
case(_site in controlsX): {
					private _roll = random 100;
					if(_roll < 10) then {
						[[_site],"A3A_fnc_LOG_Airdrop"] remoteExec ["A3A_fnc_scheduler",2];
					} else {
						[[_site],"A3A_fnc_LOG_Helicrash"] remoteExec ["A3A_fnc_scheduler", 2];
					};
                };
                default {};```

wersal454 committed:
Update Vanilla.hpp

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 committed:
Update fn_LOG_Crashsite_Satelite.sqf

wersal454 commented:
Well it is ready, except there is some weirdness with launching planes and uavs. I am inviting you to test it out.

wersal454 committed:
Update Vanilla.hpp

wersal454 committed:
Update fn_LOG_Crashsite_Satelite.sqf

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 committed:
Update fn_missionRequest.sqf

wersal454 commented:
Much better crashing path for planes and UAVs(by attaching them to quad bike(thanks Zets!)) + some minor changes. Added mission to the mission pool. Airdrop and helicrash missions now has ~35% of spawning and crashsite has 30% of spawning. (Thanks Silence! And thanks for babysitting me with my code).
Mission is probably ready for public testing.

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 committed:
Update fn_LOG_Crashsite_Satelite.sqf

Silence-112 commented:
This is continued further in #144, though supposedly works as-is in this PR

Silence-112 commented on A3A/addons/core/functions/Missions/fn_missionRequest.sqf:
```diff
@@ -193,12 +193,9 @@ switch (_type) do {
-					} else {
-						[[_site],"A3A_fnc_LOG_Helicrash"] remoteExec ["A3A_fnc_scheduler", 2];
-					};
+					private _LOGmissions = ["A3A_fnc_LOG_Airdrop", 0.35, "A3A_fnc_LOG_Helicrash", 0.35, "A3A_fnc_LOG_Crashsite", 0.3];
+					private _LOGmission = selectRandomWeighted _LOGmissions;
+					[[_site],_LOGmission] remoteExec ["A3A_fnc_scheduler", 2];
```
Use camel case for variable names, it's easier to read and follows standard conventions

```suggestion
					private _logMissions = ["A3A_fnc_LOG_Airdrop", 0.35, "A3A_fnc_LOG_Helicrash", 0.35, "A3A_fnc_LOG_Crashsite", 0.3];
					private _logMission = selectRandomWeighted _logMissions;
					[[_site],_logMission] remoteExec ["A3A_fnc_scheduler", 2];
```

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_missionRequest.sqf

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

wersal454 committed:
typo fix

wersal454 committed:
Update CfgFunctions.hpp

wersal454 committed:
Rename fn_effects to effect_crashingEffects.sqf

wersal454 committed:
Update fn_LOG_Crashsite.sqf

Silence-112 commented on A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf:
```diff
@@ -0,0 +1,721 @@
+};
+
+// selecting classnames
+private _reconvehicleClass = selectRandom ((_faction get "vehiclesPlanesTransport") + (_faction get "uavsAttack") + (_faction get "vehiclesDropPod"));
+private _pilotClass = _faction get "unitPilot";
+
+if (_reconvehicleClass in (_faction get "vehiclesDropPod") ) exitWith { 
+   [_markerX] call A3A_fnc_LOG_Crashsite_Satelite;
+};
```
```suggestion
private _reconVehicleDroppod = _faction getOrDefault ["vehiclesDropPod", []];

private _reconVehicleClass = selectRandom ((_faction get "vehiclesPlanesTransport") + (_faction get "uavsAttack") + _reconVehicleDroppod);
private _pilotClass = _faction get "unitPilot";

if (_reconVehicleClass in _reconVehicleDroppod) exitWith { 
   [_markerX] call A3A_fnc_LOG_Crashsite_Satelite;
};
```

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 committed:
Rename effect_crashingEffects.sqf to fn_effect_crashingEffects.sqf

wersal454 committed:
Update fn_LOG_Crashsite_Satelite.sqf

Silence-112 commented on A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf:
```diff
@@ -0,0 +1,721 @@
+///new
+private _infantrySquadArray = selectRandom ([_faction, "groupsTierMedium"] call SCRT_fnc_unit_flattenTier);
+
+if (isNil "_reconvehicleClass" || 
+    {isNil "_cargoTruckClass"} || 
+    {isNil "_specOpsArray"}) 
+exitWith {
+	["LOG"] remoteExec ["A3A_fnc_missionRequest",2];
+    Error("Problems with faction template, rerequesting new logistics mission.");
+};
```
This isn't that readable. It would be better to do it this way since the variable names are quite short and you're not checking much
```suggestion
if (isNil "_reconvehicleClass" || {isNil "_cargoTruckClass"} || {isNil "_specOpsArray"}) exitWith {
	["LOG"] remoteExec ["A3A_fnc_missionRequest",2];
    Error("Problems with faction template, rerequesting new logistics mission.");
};
```

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 committed:
Merge branch 'unstable' into crashsite-mission

wersal454 committed:
Update fn_effect_crashingEffects.sqf

wersal454 committed:
stringtable fix

Silence-112 commented:
After testing I have a few improvements that can be made:
> Stringtable:
- The capitalisation is off and the description of the mission is hard to understand in English (I can rewrite that if needed)

- The mission description uses the completely wrong description if arms dealer isn't present, saying a helicopter crashed.

> Mission:
- Make it clearer that you need to take the box back to HQ (If arms dealer isn't present). Can probably tie that into the mission description, so this should technically go under stringtable.

- Disable simulation on the vehicle when it "crashes".

- When arms dealer is present, the only task you get is "Take the data to the arms dealer", but it never tells you the general vicinity of the crashsite.

- When arms dealer is present and you somehow find the data, you have to take it to HQ, taking it to arms dealer won't do anything.

- Box doesn't get deleted after delivering it to hq, nor does the vehicle.

Silence-112 commented on A3A/addons/scrt/Stringtable.xml:
```diff
@@ -7783,7 +7783,7 @@
             </Key>
             <Key ID="STR_A3A_Missions_LOG_crashsite_task_desc">
-                <Original>%1 helicopter with some valueable cargo has been shot down in %2. Bring helicopter's cargo to the HQ before enemy's rescue team. We must get it before %3. Enemy frontline troops already found the crashsite and guarding it.</Original>
+                <Original>%1 The reconnaissance vehicle was shot down. Go to the supposed crash site in %2. Collect the data and bring it to the HQ before enemy rescue team. We must get it before %3. Enemy frontline troops are on their way.</Original>
```
Won't this read:
"Faction The reconnaissance vehicle was shot down."?

wersal454 replied:
probably yes

Silence-112 reviewed (commented):
Will test again when changes have been done

wersal454 committed:
checks for trader

wersal454 committed:
Update Stringtable.xml

Silence-112 commented on A3A/addons/scrt/Stringtable.xml:
```diff
@@ -7773,11 +7773,23 @@
                 <Korean>%2에서 가치 있는 화물을 적재한 %1의 헬리콥터가 격추되었습니다. 헬리콥터의 화물을 적 구조대보다 먼저 본부로 가져오십시오. %3 이전에 확보해야 합니다. 적의 최전선 부대가 이미 추락 지점을 발견하고 지키는 중입니다.</Korean>
             </Key>
+            <Key ID="STR_A3A_Missions_LOG_crashsite_task_header">
+                <Original>Reconnaissance vehicle crash site</Original>
```
```suggestion
                <Original>Reconnaissance Vehicle Crash Site</Original>
```

wersal454 committed:
Update A3A/addons/scrt/Stringtable.xml

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

Silence-112 commented on A3A/addons/scrt/Stringtable.xml:
```diff
@@ -7773,11 +7773,23 @@
+                <Russian>Транспорт %1 c разведданными был подбит в %2. Соберите их и доставьте в штаб до того как их заберёт вражеская эвакуационная команда. Мы должны сделать это %3. Вражеские передовые отряды уже в пути.</Russian>
+            </Key>
+            <Key ID="STR_A3A_Missions_LOG_crashsite_task_alt">
+                <Original>Deliver the data to arms dealer. He will pay handsomely (Petros won't be too upset).</Original>
```
```suggestion
                <Original>Deliver the crashsite data to the arms dealer. He will pay handsomely! Petros won't be too upset about it.</Original>
```

wersal454 committed:
Update A3A/addons/scrt/Stringtable.xml

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

wersal454 committed:
fixes

wersal454 committed:
fixes + trader checks

wersal454 committed:
reveal enemy markers reward

wersal454 committed:
reveal enemy markers reward

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 committed:
Update fn_LOG_Crashsite_Satelite.sqf

wersal454 committed:
Update fn_LOG_Crashsite_Satelite.sqf

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 commented:
Fixed stringtables and trader checks and made box and vehicles disappear aaand added "reveal random marker(s)" as a reward

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 committed:
Update fn_LOG_Crashsite_Satelite.sqf

wersal454 commented:
https://www.youtube.com/watch?v=X1tyybVuSBU

wersal454 committed:
typofix

wersal454 committed:
Update fn_effect_crashingEffects.sqf

wersal454 committed:
Update fn_effect_crashingEffects.sqf

wersal454 committed:
Rename sonic.ogg to Sonic.ogg

wersal454 committed:
Add files via upload

Silence-112 commented on A3A/addons/scrt/Stringtable.xml:
```diff
@@ -7773,11 +7773,23 @@
+                <Russian>Направляйтесь на предпологаемое место крушения</Russian>
+            </Key>
+            <Key ID="STR_A3A_Missions_LOG_crashsite_task_desc">
+                <Original>%1 reconnaissance vehicle was shot down. Go to the supposed crash site in %2. Collect the data and bring it to the HQ before enemy rescue team. We must get it before %3. Enemy frontline troops are on their way.</Original>
```
```suggestion
                <Original>%1s' reconnaissance vehicle was shot down. Go to the supposed crash site in %2. Collect the data and bring it back to HQ before the enemy recovery team. We must get it before %3. Enemy frontline troops are on their way to recover the data.</Original>
```

wersal454 replied:
https://www.youtube.com/watch?v=X1tyybVuSBU

wersal454 committed:
Update A3A/addons/scrt/Stringtable.xml

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

wersal454 committed:
Trader task id update

Silence-112 commented on A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf:
```diff
@@ -0,0 +1,861 @@
+
+if (!isNil "traderMarker") then { ///checking if trader is spawned
+    if (alive _box && {_box distance (getMarkerPos respawnTeamPlayer) > 50} || _box distance (getMarkerPos traderMarker) < 50) then {
+    deleteVehicle _box;
+    };
+} else {
+    if (alive _box && {_box distance (getMarkerPos respawnTeamPlayer) > 50}) then {
+    deleteVehicle _box;
```
both deleteVehicle statements need an indent, every potential future developer will hate you otherwise

Silence-112 commented on A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf:
```diff
@@ -0,0 +1,861 @@
+{[_x] spawn A3A_fnc_groupDespawner} forEach _groups;
+
+if (!isNil "traderMarker") then { ///checking if trader is spawned
+    if (alive _box && {_box distance (getMarkerPos respawnTeamPlayer) > 50} || _box distance (getMarkerPos traderMarker) < 50) then {
```
```suggestion
    if (alive _box && {_box distance (getMarkerPos respawnTeamPlayer) > 50} || {_box distance (getMarkerPos traderMarker) < 50}) then {

```

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

wersal454 committed:
Update fn_LOG_Crashsite.sqf

Silence-112 commented on A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf:
```diff
@@ -0,0 +1,861 @@
+  		    [(selectRandom [5,7])] call A3U_fnc_revealRandomZones;
+	    };
+    };
+    case(_box distance (getMarkerPos traderMarker) < 50 || _box distance (getMarkerPos traderMarker) < 50): {
```
Double check
```suggestion
    case(_box distance (getMarkerPos traderMarker) < 50): {

```

wersal454 committed:
Update fn_LOG_Crashsite_Satelite.sqf

wersal454 committed:
Update fn_LOG_Crashsite.sqf

Silence-112 commented on A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf:
```diff
@@ -0,0 +1,861 @@
+        [10*_bonus,theBoss] call A3A_fnc_addScorePlayer;
+        [250*_bonus,theBoss, true] call A3A_fnc_addMoneyPlayer;
+        ["Large", _sideX] remoteExec ["A3A_fnc_selectIntel", 2];
+        [(position _box), 4000, 1200, false] spawn SCRT_fnc_common_recon; ///params ["_position", "_radius", "_revealTime", ["_isInterrogation", false]]; ///revels enemy location only once, which is not good?, dunno maybe it should do it every 2-5 minutes
+        sleep 60;
+        [(position _box), 4000, 1200, false] spawn SCRT_fnc_common_recon;
+        sleep 60;
+        [(position _box), 4000, 1200, false] spawn SCRT_fnc_common_recon;
+        sleep 60;
+        [(position _box), 4000, 1200, false] spawn SCRT_fnc_common_recon;
+        deleteVehicle _box;
```
"What the absolute **** is this?!!?"
Sincerely,
The scheduler

This will need to be changed in some way, shape or form. It's just awful.
Perhaps use a for loop?
```suggestion
        for "_i" from 0 to 3 do {
          [(position _box), 4000, 1200, false] spawn SCRT_fnc_common_recon;
          uiSleep 60;
        };        
        deleteVehicle _box;
```
Make sure to test it before merging change

wersal454 replied:
yep, that works

Silence-112 commented on A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf:
```diff
@@ -0,0 +1,853 @@
+        sleep 60;
+        [(position _box), 4000, 1200, false] spawn SCRT_fnc_common_recon;
+        deleteVehicle _box;
+	    if (hideEnemyMarkers) then {
+  		    [(selectRandom [5,7])] call A3U_fnc_revealRandomZones;
+	    };
```
bad indentation makes me cry D:

Silence-112 commented on A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf:
```diff
@@ -0,0 +1,853 @@
+        ||
+	    _box distance (getMarkerPos respawnTeamPlayer) < 50
+        ||
+        _box distance (getMarkerPos traderMarker) < 50
```
be consistent with your format/indents
```suggestion
            _box distance (getMarkerPos traderMarker) < 50
```

Silence-112 commented on A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf:
```diff
@@ -0,0 +1,853 @@
+                        _x assignAsCargo _cargoVehicle2;
+                        [_x] join _heliInfGroup;  
+                        [_x] call A3A_fnc_NATOinit;
+                    } forEach units _cargoVehicleGroup;
```
more bad indentation, I cri

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

Silence-112 commented on A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf:
```diff
@@ -0,0 +1,853 @@
+
+    _isEnemyKnowsAboutTeamplayer = false;
+    {
+        _rebel = _x;
```
Unless there's a specific reason this isn't private, it probably should be
```suggestion
        private _rebel = _x;
```

wersal454 committed:
Update fn_LOG_Crashsite_Satelite.sqf

wersal454 committed:
Update fn_LOG_Crashsite.sqf

Silence-112 commented on A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf:
```diff
@@ -0,0 +1,853 @@
+    true
+] call BIS_fnc_taskCreate;
+[_taskId2, "LOG", "CREATED"] remoteExecCall ["A3A_fnc_taskUpdate", 2];
+};
```
Fix indentation please
```suggestion
if (!isNil "traderMarker") then { ///checking if trader is spawned
    [
        [teamPlayer,civilian],
        _taskId2,
        [format [localize "STR_A3A_Missions_LOG_crashsite_task_alt", _faction get "name", _destinationName, _displayTime], localize "STR_A3A_Missions_LOG_crashsite_task_header", _markerX],
        traderMarker,
        false,
        0,
        true,
        "whiteboard",
        true
    ] call BIS_fnc_taskCreate;
    [_taskId2, "LOG", "CREATED"] remoteExecCall ["A3A_fnc_taskUpdate", 2];
};

```

wersal454 committed:
reward update

wersal454 committed:
Merge branch 'unstable' into crashsite-mission

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

wersal454 committed:
Update A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf

Co-authored-by: Silence <78276788+SilenceIsFatto@users.noreply.github.com>

wersal454 committed:
Update fn_LOG_Crashsite.sqf

wersal454 committed:
Update fn_LOG_Crashsite_Satelite.sqf

wersal454 committed:
Merge branch 'unstable' into crashsite-mission

wersal454 committed:
Merge branch 'unstable' into crashsite-mission

wersal454 committed:
Merge branch 'unstable' into crashsite-mission

wersal454 committed:
Merge branch 'unstable' into crashsite-mission

wersal454 committed:
Merge branch 'unstable' into crashsite-mission

wersal454 committed:
Merge branch 'unstable' into crashsite-mission

Silence-112 commented:
Seems to work pretty much fine now except that delivering to arms dealer still doesn't work. Honestly, it'd be better to just remove the arms dealer task, but keep the reward upon HQ delivery if the arms dealer is present

I did experience one case of infantry spawning next to a QRF truck but never getting in it, causing the truck to just stay where it spawned. Probably a one off issue, haven't had it again

wersal454 committed:
Merge branch 'unstable' into crashsite-mission

wersal454 committed:
Merge branch 'unstable' into crashsite-mission

wersal454 committed:
bugfixes

wersal454 committed:
Merge branch 'unstable' into crashsite-mission

Silence-112 committed:
cleanup + small bugfixes

wersal454 commented:
![28cad7edc8fa911b51a8aaf5661cf61f](https://github.com/SilenceIsFatto/A3-Antistasi-Ultimate/assets/29953387/b523d9c0-80a3-4406-a67f-3329d8e0c976)

Silence-112 commented:
![elmo-fire](https://github.com/SilenceIsFatto/A3-Antistasi-Ultimate/assets/78276788/0c4cf485-edfc-4ff6-8199-0293ccff5ab7)

wersal454 committed:
quick fix

wersal454 commented:
everything appears to be working though space capsule box needs to be added to ```fn_common_moveStatic```

Silence-112 committed:
add box from satellite mission

Silence-112 committed:
Merge branch 'crashsite-mission' of https://github.com/wersal454/A3-Antistasi into pr/126
